File: RdFinance2/src/ca/mss/rd/excel/ExcelFunctions.java
package ca.mss.rd.excel;

import ca.mss.rd.util.UtilMath;

public class ExcelFunctions {

	final public static String className = ExcelFunctions.class.getName();
	final public static long serialVersionUID = className.hashCode();
	
	final public static double ZERO = 0.0099999999999999999999999999999999D;
	
	final static public double round(double value, int scale){
		return UtilMath.round(value, scale);
	}
	
	final public static double PMT(double interestRate, double principalValue, int numberOfPeriods){
		return interestRate * principalValue * Math.pow(1.0+interestRate, numberOfPeriods)/(1.0-Math.pow(1+interestRate,numberOfPeriods));
	}

}

File: RdFinanceBD/src.cpp/ExcelFunctions.cpp
#include "ExcelFunctions.h"

namespace ca
{
	namespace mss
	{
		namespace rd
		{
			namespace excel
			{

const std::wstring ExcelFunctions::className = ExcelFunctions::typeid::getName();
const java::math::BigDecimal ExcelFunctions::ZERO = java::math::BigDecimal(L"0.0");
const java::math::BigDecimal ExcelFunctions::ONE = java::math::BigDecimal(L"1.0");
const java::math::BigDecimal ExcelFunctions::HUNDRED = java::math::BigDecimal(L"100.0");

				BigDecimal ExcelFunctions::PMT(BigDecimal interestRate, BigDecimal principalValue, int numberOfPeriods)
				{
					double pow = pow(interestRate.add(ONE), numberOfPeriods);
					BigDecimal pow1 = BigDecimal(pow);
					BigDecimal pow2 = (BigDecimal(pow)).subtract(ExcelFunctions::ONE);
					return interestRate.multiply(principalValue.multiply(pow1.divide(pow2, BigDecimal::ROUND_UP)))->negate();
				}
			}
		}
	}
}
